### ISSUE TITLE
Add internal server support for Windows 64 bit

### ISSUE THREAD

**Issue by markus-wa** [overtone/overtone#447]
Heyya!

I've been trying to get Windows 64 bit working with an internal server (`overtone.live`) to get better performance when reading the buffers, but I'm having a bit of a hard time.
I know it's not officially supported right now but maybe if it's not too complicated I would like to try to fix this.

However, right now I'm sort of stuck, getting the following error when calling [this SC function](https://github.com/supercollider/supercollider/blob/9913cf8e54f56394916b7a9781a953a917ce9365/include/server/SC_WorldOptions.h#L97
):

`SCSYNTH_DLLEXPORT_C int World_CopySndBuf(struct World* world, uint32 index, struct SndBuf* outBuf, bool onlyIfChanged, bool* didChange);`

```
Exception in thread "Thread-135" java.lang.IllegalStateException: Can't autogenerate a direct buffer on memory read
        at com.sun.jna.Pointer.getValue(Pointer.java:428)
        at com.sun.jna.Structure.readField(Structure.java:732)
        at com.sun.jna.Structure.read(Structure.java:591)
        at com.sun.jna.Structure.autoRead(Structure.java:2203)
        at overtone.sc.machinery.server.native.lib_ce4ed375_6714_4b78_b528_22b6cc509d65.World_CopySndBuf(Native Method)
        at overtone.sc.machinery.server.native$eval4070$fn__4071.invoke(native.clj:36)
        at overtone.sc.machinery.server.native$scsynth_get_buffer_data.invokeStatic(native.clj:278)
        at overtone.sc.machinery.server.native$scsynth_get_buffer_data.invoke(native.clj:270)
        at overtone.sc.buffer$buffer_data.invokeStatic(buffer.clj:598)
        at overtone.sc.buffer$buffer_data.invoke(buffer.clj:591)
        at shadertone.tone$buffer_data_read.invokeStatic(tone.clj:101)
        at shadertone.tone$buffer_data_read.invoke(tone.clj:97)
        at shadertone.tone$tone_fftwave_fn.invokeStatic(tone.clj:134)
        at shadertone.tone$tone_fftwave_fn.invoke(tone.clj:105)
        at shadertone.tone$tone_default_fn.invokeStatic(tone.clj:212)
        at shadertone.tone$tone_default_fn.invoke(tone.clj:179)
        at shadertone.shader$draw.invokeStatic(shader.clj:585)
        at shadertone.shader$draw.invoke(shader.clj:548)
        at shadertone.shader$update_and_draw.invokeStatic(shader.clj:697)
        at shadertone.shader$update_and_draw.invoke(shader.clj:664)
        at shadertone.shader$run_thread.invokeStatic(shader.clj:728)
        at shadertone.shader$run_thread.invoke(shader.clj:722)
        at shadertone.shader$start_shader_display$fn__12777.invoke(shader.clj:935)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.lang.Thread.run(Thread.java:835)
```

I have spent quite a few hours trying out different things, including modifying Overtone and `clj-native` sources as well as using newer JNA versions.

So far I tried various different signatures here: https://github.com/overtone/overtone/blob/f6d414f884f1b6d3166195b49276174efddf2cf2/src/overtone/sc/machinery/server/native.clj#L176

and here (I also noticed the commented section, maybe that was from an earlier attempt?): https://github.com/overtone/overtone/blob/f6d414f884f1b6d3166195b49276174efddf2cf2/src/overtone/sc/machinery/server/native.clj#L273-L277

Has anyone seen this before or has a hint on how to proceed? Any help would be greatly appreciated 🙂.

**Comment by hlolli** [comment]
Hey!

Thanks for looking deeply into this problem. Since I'm not a windows user (and I recently deleted my VirtualBox partition), I would very much appreachiate if you can stick around and help me solve this.

For a quick fix, I'd either try to use older scsynth version, this can be overriden with your dependency tool (lein or tools-deps-alpha), and/or use older version of java-jna.

There is clearly a problem with java-jna and out mapping here. I'd give everything a try. What Overtone needs is better communication with native libraries. There are multiple problems now we are having with the internal supercollider. A new jna binding library from `techascent/tech.jna` is worth looking into. Please keep reminding me here about it, and tell me about your progress if you have a breaktrough. If you're interested to help me implementing new jna bindings/push newer scsynth binaries or help in any way, please let me know, the Windows department is only me atm (and I don't have windows :) )

**Comment by markus-wa** [comment]
Hey @hlolli - I'm a bit busy since I'll be moving countries over the next two months, but I will try some more things if I find some time - honestly though I'm also switching over to mainly using Linux once I no longer have my desktop setup at my current place.

I can't make any promises right now as I'd also like to get https://github.com/overtone/shadertone/issues/35 figured out first, but if I find some time I will try some things out.

I'll put this into my TODO list, hopefully we can figure it out.

**Comment by rogerallen** [comment]
@markus-wa do you have docs/links for the steps you took to get the Win10 build going?

**Comment by markus-wa** [comment]
@rogerallen I wrote it down in the readme of my 'scratch' project with shadertone: https://github.com/markus-wa/av-clj (note that you'll need my shadertone branch for this one) which will take your main output as input for shadertone, but I'm not sure it's supported for all soundcards.

To get overtone alone working you might just need the supercollider bit from the project above. Note that you should set maxLogins to 100 or so instead of 2, else you constantly have to `Server.default.reboot` if something breaks (which regularly happens to me).

Then you should be able to:
- `lein repl`
- `(use 'overtone.core)`
- `(connect-external-server)`

PS: this is of course just for external SC servers, while this issue here is about internal servers.

**Comment by markus-wa** [comment]
Ahhh sorry, I thin I misunderstood. You're talking about *building* overtone, not running it.

I'm not sure I ever built overtone on Windows, only shadertone. But I will verify this today.

The few things I did end up tryi g on overtone might have all been on linux.

Will update in a bit.

**Comment by rogerallen** [comment]
I'm just trying to understand what you did to get to the point where you filed this bug.

**Comment by markus-wa** [comment]
I've tried really hard to reproduce the original error in this issue but had no luck. Unfortunately I no longer have access to the machine I originally did this on (and it has probably been wiped anyway).

Now I'm just getting this error: `INFO: Running dep handler: :overtone.sc.bus/allocate-hw-audio-busses
java.util.concurrent.TimeoutException: deref! timeout error. Dereference took longer than 5000 ms whilst attempting to fetch server information` which seems to be earlier on in the startup process.

<details>
<summary>full output</summary>
```
$ lein run
--> Loading Overtone...
--> Booting internal SuperCollider server...
Mar 22, 2020 1:40:33 PM clojure.lang.Reflector invokeMatchingMethod
INFO: booting internal audio server
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: The internal scsynth server has booted...
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: satisfying deps: (:internal-server-booted)
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: Running dep handler: :overtone.sc.machinery.server.connection/connect-internal
--> Connecting to internal SuperCollider server...
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: deps-satisfied: #{:internal-server-booted}
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: satisfying deps: (:server-connected)
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: Running dep handler: :overtone.sc.machinery.synthdef/load-all-synthdefs
--> Connection established
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: satisfying deps: (:synthdefs-loaded)
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: Running dep handler: :overtone.sc.foundation-groups/setup-foundation-groups
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: satisfying deps: (:foundation-groups-created)
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: deps-satisfied: #{:internal-server-booted :server-connected}
Mar 22, 2020 1:40:34 PM clojure.lang.Reflector invokeMatchingMethod
INFO: Running dep handler: :overtone.sc.bus/allocate-hw-audio-busses
java.util.concurrent.TimeoutException: deref! timeout error. Dereference took longer than 5000 ms whilst attempting to fetch server information
    at overtone.helpers.lib$deref_BANG_.invokeStatic(lib.clj:275)
    at overtone.helpers.lib$deref_BANG_.invoke(lib.clj:261)
    at overtone.helpers.lib$deref_BANG_.invokeStatic(lib.clj:269)
    at overtone.helpers.lib$deref_BANG_.invoke(lib.clj:261)
    at overtone.sc.info$server_info.invokeStatic(info.clj:73)
    at overtone.sc.info$server_info.invoke(info.clj:39)
    at overtone.sc.info$server_num_input_buses.invokeStatic(info.clj:130)
    at overtone.sc.info$server_num_input_buses.invoke(info.clj:123)
    at overtone.sc.bus$allocate_hw_audio_buses.invokeStatic(bus.clj:346)
    at overtone.sc.bus$allocate_hw_audio_buses.invoke(bus.clj:344)
    at overtone.libs.deps$satisfy_STAR_$execute_tasks__1854$fn__1861.invoke(deps.clj:85)
    at overtone.libs.deps$satisfy_STAR_$execute_tasks__1854.invoke(deps.clj:84)
    at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
    at clojure.core$reduce.invokeStatic(core.clj:6827)
    at clojure.core$reduce.invoke(core.clj:6810)
    at overtone.libs.deps$satisfy_STAR_.invokeStatic(deps.clj:95)
    at overtone.libs.deps$satisfy_STAR_.invoke(deps.clj:77)
    at clojure.core$binding_conveyor_fn$fn__5739.invoke(core.clj:2036)
    at clojure.lang.AFn.applyToHelper(AFn.java:156)
    at clojure.lang.RestFn.applyTo(RestFn.java:132)
    at clojure.lang.Agent$Action.doRun(Agent.java:114)
    at clojure.lang.Agent$Action.run(Agent.java:163)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.base/java.lang.Thread.run(Thread.java:834)
Mar 22, 2020 1:40:39 PM clojure.lang.Reflector invokeMatchingMethod
SEVERE: Exception in dependency handler: :overtone.sc.bus/allocate-hw-audio-busses

Mar 22, 2020 1:40:39 PM clojure.lang.Reflector invokeMatchingMethod
INFO: deps-satisfied: #{:synthdefs-loaded :internal-server-booted :server-connected}
Mar 22, 2020 1:40:39 PM clojure.lang.Reflector invokeMatchingMethod
INFO: Running dep handler: :overtone.sc.bus/create-monitor-group
Mar 22, 2020 1:40:39 PM clojure.lang.Reflector invokeMatchingMethod
INFO: Running dep handler: :overtone.sc.clock/start-clock
Mar 22, 2020 1:40:39 PM clojure.lang.Reflector invokeMatchingMethod
INFO: Running dep handler: :overtone.studio.mixer/start-bus-mixers
java.util.concurrent.TimeoutException: deref! timeout error. Dereference took longer than 5000 ms whilst attempting to fetch server information
    at overtone.helpers.lib$deref_BANG_.invokeStatic(lib.clj:275)
    at overtone.helpers.lib$deref_BANG_.invoke(lib.clj:261)
    at overtone.helpers.lib$deref_BANG_.invokeStatic(lib.clj:269)
    at overtone.helpers.lib$deref_BANG_.invoke(lib.clj:261)
    at overtone.sc.info$server_info.invokeStatic(info.clj:73)
    at overtone.sc.info$server_info.invoke(info.clj:39)
    at overtone.sc.info$server_num_input_buses.invokeStatic(info.clj:130)
    at overtone.sc.info$server_num_input_buses.invoke(info.clj:123)
    at overtone.studio.mixer$start_io_mixers$fn__10488.invoke(mixer.clj:92)
    at overtone.sc.machinery.server.comms$with_server_sync.invokeStatic(comms.clj:117)
    at overtone.sc.machinery.server.comms$with_server_sync.invoke(comms.clj:99)
    at overtone.studio.mixer$start_io_mixers.invokeStatic(mixer.clj:91)
    at overtone.studio.mixer$start_io_mixers.invoke(mixer.clj:88)
    at overtone.libs.deps$satisfy_STAR_$execute_tasks__1854$fn__1861.invoke(deps.clj:85)
    at overtone.libs.deps$satisfy_STAR_$execute_tasks__1854.invoke(deps.clj:84)
    at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
    at clojure.core$reduce.invokeStatic(core.clj:6827)
    at clojure.core$reduce.invoke(core.clj:6810)
    at overtone.libs.deps$satisfy_STAR_.invokeStatic(deps.clj:95)
    at overtone.libs.deps$satisfy_STAR_.invoke(deps.clj:77)
    at clojure.core$binding_conveyor_fn$fn__5739.invoke(core.clj:2036)
    at clojure.lang.AFn.applyToHelper(AFn.java:156)
    at clojure.lang.RestFn.applyTo(RestFn.java:132)
    at clojure.lang.Agent$Action.doRun(Agent.java:114)
    at clojure.lang.Agent$Action.run(Agent.java:163)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.base/java.lang.Thread.run(Thread.java:834)
Mar 22, 2020 1:40:44 PM clojure.lang.Reflector invokeMatchingMethod
SEVERE: Exception in dependency handler: :overtone.studio.mixer/start-bus-mixers

Mar 22, 2020 1:40:44 PM clojure.lang.Reflector invokeMatchingMethod
INFO: deps-satisfied: #{:synthdefs-loaded :foundation-groups-created :internal-server-booted :server-connected}
Exception in thread "main" Syntax error compiling at (overtone/live.clj:6:1).
    at clojure.lang.Compiler.load(Compiler.java:7647)
    at clojure.lang.RT.loadResourceScript(RT.java:381)
    at clojure.lang.RT.loadResourceScript(RT.java:372)
    at clojure.lang.RT.load(RT.java:463)
    at clojure.lang.RT.load(RT.java:428)
    at clojure.core$load$fn__6824.invoke(core.clj:6126)
    at clojure.core$load.invokeStatic(core.clj:6125)
    at clojure.core$load.doInvoke(core.clj:6109)
    at clojure.lang.RestFn.invoke(RestFn.java:408)
    at clojure.core$load_one.invokeStatic(core.clj:5908)
    at clojure.core$load_one.invoke(core.clj:5903)
    at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
    at clojure.core$load_lib.invokeStatic(core.clj:5947)
    at clojure.core$load_lib.doInvoke(core.clj:5928)
    at clojure.lang.RestFn.applyTo(RestFn.java:142)
    at clojure.core$apply.invokeStatic(core.clj:667)
    at clojure.core$load_libs.invokeStatic(core.clj:5985)
    at clojure.core$load_libs.doInvoke(core.clj:5969)
    at clojure.lang.RestFn.applyTo(RestFn.java:137)
    at clojure.core$apply.invokeStatic(core.clj:669)
    at clojure.core$use.invokeStatic(core.clj:6093)
    at clojure.core$use.doInvoke(core.clj:6093)
    at clojure.lang.RestFn.invoke(RestFn.java:408)
    at av_clj.loader$eval163$loading__6706__auto____164.invoke(loader.clj:1)
    at av_clj.loader$eval163.invokeStatic(loader.clj:1)
    at av_clj.loader$eval163.invoke(loader.clj:1)
    at clojure.lang.Compiler.eval(Compiler.java:7176)
    at clojure.lang.Compiler.eval(Compiler.java:7165)
    at clojure.lang.Compiler.load(Compiler.java:7635)
    at clojure.lang.RT.loadResourceScript(RT.java:381)
    at clojure.lang.RT.loadResourceScript(RT.java:372)
    at clojure.lang.RT.load(RT.java:463)
    at clojure.lang.RT.load(RT.java:428)
    at clojure.core$load$fn__6824.invoke(core.clj:6126)
    at clojure.core$load.invokeStatic(core.clj:6125)
    at clojure.core$load.doInvoke(core.clj:6109)
    at clojure.lang.RestFn.invoke(RestFn.java:408)
    at clojure.core$load_one.invokeStatic(core.clj:5908)
    at clojure.core$load_one.invoke(core.clj:5903)
    at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
    at clojure.core$load_lib.invokeStatic(core.clj:5947)
    at clojure.core$load_lib.doInvoke(core.clj:5928)
    at clojure.lang.RestFn.applyTo(RestFn.java:142)
    at clojure.core$apply.invokeStatic(core.clj:667)
    at clojure.core$load_libs.invokeStatic(core.clj:5985)
    at clojure.core$load_libs.doInvoke(core.clj:5969)
    at clojure.lang.RestFn.applyTo(RestFn.java:137)
    at clojure.core$apply.invokeStatic(core.clj:669)
    at clojure.core$use.invokeStatic(core.clj:6093)
    at clojure.core$use.doInvoke(core.clj:6093)
    at clojure.lang.RestFn.invoke(RestFn.java:421)
    at av_clj.core$eval157$loading__6706__auto____158.invoke(core.clj:1)
    at av_clj.core$eval157.invokeStatic(core.clj:1)
    at av_clj.core$eval157.invoke(core.clj:1)
    at clojure.lang.Compiler.eval(Compiler.java:7176)
    at clojure.lang.Compiler.eval(Compiler.java:7165)
    at clojure.lang.Compiler.load(Compiler.java:7635)
    at clojure.lang.RT.loadResourceScript(RT.java:381)
    at clojure.lang.RT.loadResourceScript(RT.java:372)
    at clojure.lang.RT.load(RT.java:463)
    at clojure.lang.RT.load(RT.java:428)
    at clojure.core$load$fn__6824.invoke(core.clj:6126)
    at clojure.core$load.invokeStatic(core.clj:6125)
    at clojure.core$load.doInvoke(core.clj:6109)
    at clojure.lang.RestFn.invoke(RestFn.java:408)
    at clojure.core$load_one.invokeStatic(core.clj:5908)
    at clojure.core$load_one.invoke(core.clj:5903)
    at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
    at clojure.core$load_lib.invokeStatic(core.clj:5947)
    at clojure.core$load_lib.doInvoke(core.clj:5928)
    at clojure.lang.RestFn.applyTo(RestFn.java:142)
    at clojure.core$apply.invokeStatic(core.clj:667)
    at clojure.core$load_libs.invokeStatic(core.clj:5985)
    at clojure.core$load_libs.doInvoke(core.clj:5969)
    at clojure.lang.RestFn.applyTo(RestFn.java:137)
    at clojure.core$apply.invokeStatic(core.clj:667)
    at clojure.core$require.invokeStatic(core.clj:6007)
    at clojure.core$require.doInvoke(core.clj:6007)
    at clojure.lang.RestFn.invoke(RestFn.java:408)
    at user$eval140$fn__144.invoke(form-init4001864515911791430.clj:1)
    at user$eval140.invokeStatic(form-init4001864515911791430.clj:1)
    at user$eval140.invoke(form-init4001864515911791430.clj:1)
    at clojure.lang.Compiler.eval(Compiler.java:7176)
    at clojure.lang.Compiler.eval(Compiler.java:7166)
    at clojure.lang.Compiler.load(Compiler.java:7635)
    at clojure.lang.Compiler.loadFile(Compiler.java:7573)
    at clojure.main$load_script.invokeStatic(main.clj:452)
    at clojure.main$init_opt.invokeStatic(main.clj:454)
    at clojure.main$init_opt.invoke(main.clj:454)
    at clojure.main$initialize.invokeStatic(main.clj:485)
    at clojure.main$null_opt.invokeStatic(main.clj:519)
    at clojure.main$null_opt.invoke(main.clj:516)
    at clojure.main$main.invokeStatic(main.clj:598)
    at clojure.main$main.doInvoke(main.clj:561)
    at clojure.lang.RestFn.applyTo(RestFn.java:137)
    at clojure.lang.Var.applyTo(Var.java:705)
    at clojure.main.main(main.java:37)
Caused by: java.lang.Exception: The following deps took too long (20 seconds) to be satisfied: :server-ready
    at overtone.libs.deps$wait_until_deps_satisfied.invokeStatic(deps.clj:184)
    at overtone.libs.deps$wait_until_deps_satisfied.invoke(deps.clj:168)
    at overtone.libs.deps$wait_until_deps_satisfied.invokeStatic(deps.clj:174)
    at overtone.libs.deps$wait_until_deps_satisfied.invoke(deps.clj:168)
    at overtone.sc.machinery.server.connection$boot$fn__4185.invoke(connection.clj:378)
    at overtone.sc.machinery.server.connection$boot.invokeStatic(connection.clj:363)
    at overtone.sc.machinery.server.connection$boot.invoke(connection.clj:350)
    at overtone.sc.machinery.server.connection$boot.invokeStatic(connection.clj:361)
    at overtone.sc.machinery.server.connection$boot.invoke(connection.clj:350)
    at overtone.sc.machinery.server.connection$boot.invokeStatic(connection.clj:359)
    at overtone.sc.machinery.server.connection$boot.invoke(connection.clj:350)
    at overtone.sc.server$boot_server.invokeStatic(server.clj:148)
    at overtone.sc.server$boot_server.invoke(server.clj:145)
    at overtone.studio.mixer$boot_server_and_mixer.invokeStatic(mixer.clj:194)
    at overtone.studio.mixer$boot_server_and_mixer.invoke(mixer.clj:189)
    at overtone.live$eval12416.invokeStatic(live.clj:8)
    at overtone.live$eval12416.invoke(live.clj:6)
    at clojure.lang.Compiler.eval(Compiler.java:7176)
    at clojure.lang.Compiler.load(Compiler.java:7635)
    ... 96 more
```
<details>

this happens when I do:
```sh
git clone git@github.com:markus-wa/av-clj.git
git checkout windows-internal-server
lein run
```

**Comment by markus-wa** [comment]
I just noticed this bit in the overtone.log:

```
{:level :debug,
 :timestamp 1584885279662,
 :date #inst "2020-03-22T13:54:39.662-00:00",
 :msg
 "Registering async event handler:: /server-info with key: :overtone.sc.info/get-server-info_G__12439"}
{:level :debug,
 :timestamp 1584885279663,
 :date #inst "2020-03-22T13:54:39.663-00:00",
 :msg
 "Sending: /s_new, args: [\"overtone.sc.4a6/snd-server-info\" 11 1 7 \"response-id\" 0.0]"}
{:level :error,
 :timestamp 1584885284673,
 :date #inst "2020-03-22T13:54:44.673-00:00",
 :msg
 "Exception in dependency handler: :overtone.sc.bus/allocate-hw-audio-busses\n"}
```

I will try to compare this with what I'm getting on linux if I can get that to work, especially ` "Sending: /s_new, args: [\"overtone.sc.4a6/snd-server-info\" 11 1 7 \"response-id\" 0.0]"}`

**Comment by markus-wa** [comment]
On Linux I'm seeing exactly the same messages minus the error:

```
{:level :debug,
 :timestamp 1585435579748,
 :date #inst "2020-03-28T22:46:19.748-00:00",
 :msg
 "Registering async event handler:: /server-info with key: :overtone.sc.info/get-server-info_G
__12501"}
{:level :debug,
 :timestamp 1585435579750,
 :date #inst "2020-03-28T22:46:19.750-00:00",
 :msg
 "Sending: /s_new, args: [\"overtone.sc.4a6/snd-server-info\" 11 1 7 \"response-id\" 0.0]"}
{:level :debug,
 :timestamp 1585435579792,
 :date #inst "2020-03-28T22:46:19.792-00:00",
 :msg
 "node-created: 11\nsynth-node: overtone.sc.node.SynthNode@38575d2e"}
```

So I'm not sure what to make o it. It seems like the server-info is sent but never arrives.

**Comment by markus-wa** [comment]
I've made an attempt to fix this in #454. I'm a bit worried that it might break things on OSX, maybe we'll have to add different behaviour for the two OSs.

**Comment by plexus** [comment]
Since the last two versions we no longer have an internal server, see https://groups.google.com/g/overtone/c/qndjDV5FS9Y for a write-up of why. The short version is that it was too much maintenance to be worth it.

So I think we can safely close it. Installing supercollider on Windows and making sure overtone can find `scsynth` on the path should be enough to make things work.
